core/jobs/batch_jobs/exp_validation_jobs_test.py

```diff
@@ -0,0 +1,394 @@
+                'ExampleParamOne': self.PARAM_SPECS,
+                'ExampleParamTwo': self.PARAM_SPECS
+            },
+            param_changes=self.PARAM_CHANGES_2,
+            auto_tts_enabled=feconf.DEFAULT_AUTO_TTS_ENABLED,
+            correctness_feedback_enabled=False,
+            states={
+                feconf.DEFAULT_INIT_STATE_NAME: self.STATE_3,
+                'second state': self.STATE_4,
+                'state 1': self.INVALID_STATE_1.to_dict(),
+                'state 2': self.INVALID_STATE_2.to_dict(),
+            }
+        )
+
+        # Valid opportunity model.
+        self.opportunity_1 = self.create_model(
+            opportunity_models.ExplorationOpportunitySummaryModel,
+            id=self.EXPLORATION_ID_1,
+            topic_id='topic_id1',
+            topic_name='a_topic name',
+            story_id='story_id1',
+            story_title='A story title',
+            chapter_title='A chapter title',
+            content_count=20,
+            incomplete_translation_language_codes=['hi', 'ar'],
+            translation_counts={},
+            language_codes_needing_voice_artists=['en'],
+            language_codes_with_assigned_voice_artists=[]
+        )
+
+        # Valid opportunity model.
+        self.opportunity_2 = self.create_model(
+            opportunity_models.ExplorationOpportunitySummaryModel,
+            id=self.EXPLORATION_ID_2,
+            topic_id='topic_id2',
+            topic_name='a_topic name',
+            story_id='story_id2',
+            story_title='A story title',
+            chapter_title='A chapter title',
+            content_count=20,
+            incomplete_translation_language_codes=['hi', 'ar'],
+            translation_counts={},
+            language_codes_needing_voice_artists=['en'],
+            language_codes_with_assigned_voice_artists=[]
+        )
+
+        # Valid opportunity model.
+        self.opportunity_3 = self.create_model(
+            opportunity_models.ExplorationOpportunitySummaryModel,
+            id=self.EXPLORATION_ID_3,
+            topic_id='topic_id3',
+            topic_name='a_topic name',
+            story_id='story_id3',
+            story_title='A story title',
+            chapter_title='A chapter title',
+            content_count=20,
+            incomplete_translation_language_codes=['hi', 'ar'],
+            translation_counts={},
+            language_codes_needing_voice_artists=['en'],
+            language_codes_with_assigned_voice_artists=[]
+        )
+
+    def test_run_with_no_models(self) -> None:
+        self.assert_job_output_is([])
+
+    def test_run_with_single_valid_model(self) -> None:
+        self.put_multi([self.exp_1, self.opportunity_1])
+        self.assert_job_output_is([
+            job_run_result.JobRunResult.as_stdout('EXPS SUCCESS: 1'),
+            job_run_result.JobRunResult.as_stdout('CURATED EXPS SUCCESS: 1')
+        ])
+
+    def test_run_with_single_invalid_model(self) -> None:
```

Comment:
Maybe have a separate test for each invalidity type?

Reply:
Done